exonum-light-client/src/main/java/com/exonum/client/ExonumClient.java

```diff
@@ -77,6 +80,92 @@
    */
   Optional<TransactionResponse> getTransaction(HashCode id);
 
+  /**
+   * Returns the <em>blockchain height</em> which is the height of the latest committed block
+   * in the blockchain. The block height is a distance between the last block
+   * and the "genesis", or initial, block. Therefore, the blockchain height is equal to the number
+   * of blocks plus one.
+   *
+   * <p>For example, the "genesis" block has height {@code h = 0}. The latest committed block
+   * has height {@code h = getBlockHashes().size() - 1}.
+   *
+   * @throws RuntimeException if the client is unable to complete a request
+   *        (e.g., in case of connectivity problems)
+   */
+  long getBlockchainHeight();
+
+  /**
+   * Returns the information about the block with transaction hashes included at this block.
+   * @param height blockchain height starting from 0 (genesis block)
+   * @return block information response
+   * @throws RuntimeException if the client is unable to complete a request
+   *        (e.g., in case of connectivity problems)
+   * @throws RuntimeException if block is not found by the requested height,
+   *        i.e. the requested height is greater then actual blockchain height
```

Comment:
Actually, we throw `IllegalArgumentException` in the `Blockchain`. 
The problem in this case, that we do not know actual height on the client side and we need to execute the request.  
But I agree to throw `IndexOutOfBounds` in both cases, so we can handle 404 in this way. @dmitry-timofeev WDYT?

Reply:
> Actually, we throw `IllegalArgumentException` in the `Blockchain`.

Where? We throw that in case of invalid height in `Blockchain#getBlockTransactions`. But this method is `getBlockByHeight` so I looked at `Blockchain#getBlock(long height)` and there it throws `@throws IndexOutOfBoundsException if the height is not valid` - it checks `if (height < 0 || height > blockchainHeight)`
But honestly it doesn't really matter and `IllegalArgumentException` works as well.

Reply:
Agree. But we can't get the actual height without querying the API. And it'll be good to have some informative error message in case we throw out of bounds. Something like "the height is out of range [0, N]". So we can do here:
1) left it as is 😼 
2) Put `IndexOutOfBoundsException` and trust the API and believe that 404 can be only when we send height > blockchain_height
3) If the API returns 404 then query the API once more for actual height and place it in the error message
🤔

Reply:
created ECR-2996 for this issue